test/RelayHubGas.test.ts

```diff
@@ -0,0 +1,1471 @@
+    let relayHubInstance: RelayHubInstance;
+    let recipientContract: TestRecipientInstance;
+    let verifierContract: TestVerifierEverythingAcceptedInstance;
+    let deployVerifierContract: TestDeployVerifierEverythingAcceptedInstance;
+    let forwarderInstance: IForwarderInstance;
+    let target: string;
+    let verifier: string;
+    let forwarder: string;
+    let gaslessAccount: AccountKeypair;
+    const gasLimit = '3000000';
+    const gasPrice = '1';
+    let sharedRelayRequestData: RelayRequest;
+    let env: Environment;
+    let token: TestTokenInstance;
+    let factory: SmartWalletFactoryInstance;
+
+    describe('relayCall', function () {
+        beforeEach(async function () {
+            env = await getTestingEnvironment();
+            chainId = env.chainId;
+
+            penalizer = await Penalizer.new();
+            relayHubInstance = await deployHub(penalizer.address);
+            verifierContract = await TestVerifierEverythingAccepted.new();
+            deployVerifierContract =
+                await TestDeployVerifierEverythingAccepted.new();
+            gaslessAccount = await getGaslessAccount();
+
+            const smartWalletTemplate: SmartWalletInstance =
+                await SmartWallet.new();
+            factory = await createSmartWalletFactory(smartWalletTemplate);
+            recipientContract = await TestRecipient.new();
+            const testToken = artifacts.require('TestToken');
+            token = await testToken.new();
+            target = recipientContract.address;
+            verifier = verifierContract.address;
+            relayHub = relayHubInstance.address;
+
+            forwarderInstance = await createSmartWallet(
+                _,
+                gaslessAccount.address,
+                factory,
+                gaslessAccount.privateKey,
+                chainId
+            );
+            forwarder = forwarderInstance.address;
+            await token.mint('1000', forwarder);
+
+            sharedRelayRequestData = {
+                request: {
+                    relayHub: relayHub,
+                    to: target,
+                    data: '',
+                    from: gaslessAccount.address,
+                    nonce: (await forwarderInstance.nonce()).toString(),
+                    value: '0',
+                    gas: gasLimit,
+                    tokenContract: token.address,
+                    tokenAmount: '1',
+                    tokenGas: '50000'
+                },
+                relayData: {
+                    gasPrice,
+                    relayWorker,
+                    callForwarder: forwarder,
+                    callVerifier: verifier
+                }
+            };
+        });
+
+        context('with staked and registered relay', function () {
+            const url = 'http://relay.com';
+            const message = 'Enveloping RelayHub';
+            const messageWithNoParams = 'Method with no parameters';
```

Comment:
messageWithNoParams -> variable is not being used

Reply:
fixed